### ISSUE TITLE
Episode 4: Networks are like Onions

### ISSUE THREAD

**Issue by dafnevk** [carpentries-incubator/deep-learning-intro#45]
## Learning objectives 
- Understand why convolutional and pooling layers are useful for image data 
- Use normalization as preparation step for deep learning 
- Implement a convolutional neural network on some image dataset (mnist / cifar10, stl10) 
- Plot and interpret the training process 
- Do a visual inspection of the results of your neural network 
- Understand strategies to improve your model based on the plots 
- Use drop-out layer to prevent overfitting 

Motivation: want to apply on image data, which layers are common to use in vision tasks.

**Comment by HaoZeke** [comment]
More of a comment, but many learners might not have any understanding of the way images are interpreted, perhaps a lesson on image data, channels, and simple RGB transforms should come before this.

**Comment by dafnevk** [comment]
Good point that learners might not be familiar with this. A complete lesson or even episode might be a bit too much to require for this course, but we should make sure that this episode gives enough context.

**Comment by dafnevk** [comment]
I experimented a bit with CIFAR10, based on the [Keras tutorial](https://www.tensorflow.org/tutorials/images/cnn) (see [this notebook](https://github.com/escience-academy/deep-learning-material/blob/master/notebooks/Episode4-CNN-cifar10.ipynb)). Training this model on the complete train set does not result in overfitting, so then there's not a good reason to introduce dropout. If I only take 10% of the train set, this model does overfit and dropout helps a little bit. The plus side of this approach is that training is really fast on a small dataset.

[Another Keras tutorial](https://www.tensorflow.org/tutorials/images/classification) uses a flower classification dataset, with fewer (but larger) images, that shows clear overfitting. They introduce data augmentation and dropout as solutions. This dataset is a bit less straightforward to load, the tutorial uses [image_dataset_from_directory](https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image_dataset_from_directory) but that's only available from tensorflow version 2.3.0.

**Comment by svenvanderburg** [comment]
@dafnevk both solutions sound sane to me. To solve the problem with the flower classification dataset you could prepare the dataset yourself. And using tensorflow>2.3.0 seems also like a reasonable solution if you specify this in the setup instructions.
